### ISSUE TITLE
Blank search results list, despite finding matches

### ISSUE THREAD

**Issue by tigersof** [NiklasGollenstede/tst-search#34]
Hi,

I've been using this extension quite a while now, and up until now it has been working as intended (as far as I can see!). I've suddenly encountered a problem where when entering a search term, results in a blank list despite recognising and indicating the expected X number of matches. It is therefore currently unusable.

Any ideas what could be causing this, and is there an easy fix for it?

**Comment by piroor** [comment]
@NiklasGollenstede TST 4.0 and later generates HTML elements only near the viewport, so showing/hiding search results with a CSS hack is not effective anymore. Instead we need to show/hide tabs via WebExtensions API.
Here is an experimental branch for that, but I know it is not good. I hope it helps your development.

https://github.com/piroor/tst-search/tree/support-tst-4
https://github.com/piroor/tst-search/commit/20188d23a1f511e1f1bbc0a2921cb5e75c0e650a

**Comment by fffmmm** [comment]
Yes, this is sad. TST Search only shows matching results of pinned tabs, non-pinned tab match results are not returned, however the match total is correct seems to include the tabs that are not shown.

**Comment by miltuss** [comment]
> @NiklasGollenstede TST 4.0 and later generates HTML elements only near the viewport, so showing/hiding search results with a CSS hack is not effective anymore. Instead we need to show/hide tabs via WebExtensions API. Here is an experimental branch for that, but I know it is not good. I hope it helps your development.
> 
> https://github.com/piroor/tst-search/tree/support-tst-4 [piroor@20188d2](https://github.com/piroor/tst-search/commit/20188d23a1f511e1f1bbc0a2921cb5e75c0e650a)

Hello, this patch is not working for me. Can you provide the .xpi file to repair TST-search?

**Comment by piroor** [comment]
I've pushed some more fixes to the experimental branch.

And here is the built zip, which can be loaded as a temporary addon via about:debugging.
[tst-search-0.0.7+-firefox.zip](https://github.com/user-attachments/files/16990757/tst-search-0.0.7%2B-firefox.zip)

**Comment by mzso** [comment]
@piroor 
> I've pushed some more fixes to the experimental branch.
> 
> And here is the built zip, which can be loaded as a temporary addon via about:debugging. [tst-search-0.0.7+-firefox.zip](https://github.com/user-attachments/files/16990757/tst-search-0.0.7%2B-firefox.zip)

Thanks! It works, however with very low performance.

**Comment by mzso** [comment]
I don't get why it results in ungodly amount of gigabyte IO reads and browser slowdown/unusability for 5 minutes:

<img width="1078" height="383" alt="Image" src="https://github.com/user-attachments/assets/8325b24c-a972-45db-98ca-1ad82fa147e6" />

There's also a great increase in memory usage, and then drop:

<img width="607" height="520" alt="Image" src="https://github.com/user-attachments/assets/a6236b7b-6560-43fc-ba76-5b5ebfd5b955" />
